helix-term/src/ui/mod.rs

```diff
@@ -268,15 +285,49 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> FilePi
         }
     }
     if hit_timeout {
```

Comment:
A possible alternative would be to drop this timeout to build and return an (empty) `Picker` as fast as possible, and always populate it from a background thread. I'm not entirely sure why this behaviour is preferred. On my Linux machine, the delay of spawning and scheduling a new thread is a few orders of magnitude lower than the current timeout duration, but, perhaps, on other platforms it is preferable to avoid creating a new thread if traversing the file tree is faster than what a human perceives to be a slow UI.